src/Arcus.EventGrid.Publishing/EventGridPublisher.cs

```diff
@@ -66,9 +66,10 @@ internal EventGridPublisher(Uri topicEndpoint, string authenticationKey, Policy
         /// <param name="eventId">Id of the event</param>
         /// <param name="eventType">Type of the event</param>
         /// <param name="eventBody">Body of the event</param>
-        public async Task PublishRawAsync(string eventId, string eventType, string eventBody)
+        /// <param name="eventSchema">The schema in which the event should be published.</param>
+        public async Task PublishRawAsync(string eventId, string eventType, string eventBody, EventSchema eventSchema = EventSchema.EventGrid)
```

Comment:
Do we use that or `PublishRawEventGridEventAsync` and `PublishRawCloudEventAsync`?

Reply:
I think the API on this would be even more complex if we would do that.

Reply:
Can you give an example please? (other than the event source)

Reply:
Don't know what you mean.
What I meant was that if we introduce another type in the signature, it would become more complex.

Now, we already have `PublishRaw/Many`, `Publish/Many (custom)`, `Publish/Many(cloud)`.

Reply:
My suggestion would remove the current publish raw, not the others

Reply:
Hmm, question, what do we do with the `PublishRawAsync(RawEvent)` ones? They also have to be converted to cloud events but don't have a required `source`. 😞 .

Reply:
That's what we've discussed yesterday where I would remove PublishRawAsync and replace it with PublishRawAsEventGridAsync & PublishRawAsCloudEventAsync so you can add required parameters

Reply:
Hmm, I can't even get the existing integration tests to work locally 😭 . A little hard to add some CloudEvent tests as well.